pkg/gardenlet/operation/botanist/advertisedaddresses_test.go

```diff
@@ -195,10 +214,126 @@ var _ = Describe("AdvertisedAddresses", func() {
 			}
 			botanist.APIServerAddress = "bar.foo"
 
-			addresses, err := botanist.ToAdvertisedAddresses()
+			addresses, err := botanist.ToAdvertisedAddresses(ctx)
 			Expect(err).To(HaveOccurred())
 			Expect(err.Error()).To(Equal("shoot requires managed issuer, but gardener does not have shoot service account hostname configured"))
 			Expect(addresses).To(BeNil())
 		})
 	})
+
+	Describe("#GetIngressAdvertisedEndpoints", func() {
+		It("returns none with no ingress resources", func() {
+			items, err := botanist.GetIngressAdvertisedEndpoints(ctx)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(items).To(HaveLen(0))
+		})
+
+		It("returns none when no ingress is labeled", func() {
+			// Resource does not have the expected labels
+			ingress := &networkingv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "ingress-1",
+					Namespace: shootNamespace.Name,
+				},
+				Spec: networkingv1.IngressSpec{
+					TLS: []networkingv1.IngressTLS{
+						{
+							Hosts: []string{"foo.example.org"},
+						},
+					},
+				},
+			}
+
+			Expect(botanist.SeedClientSet.Client().Create(ctx, ingress)).To(Succeed())
+			items, err := botanist.GetIngressAdvertisedEndpoints(ctx)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(items).To(HaveLen(0))
+			Expect(botanist.SeedClientSet.Client().Delete(ctx, ingress)).To(Succeed())
+		})
+
+		It("returns valid endpoints from ingress resources", func() {
+			ingressA := &networkingv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "ingress-1",
+					Namespace: shootNamespace.Name,
+					Labels: map[string]string{
+						v1beta1constants.LabelShootEndpointAdvertise: "true",
+					},
+				},
+				Spec: networkingv1.IngressSpec{
+					TLS: []networkingv1.IngressTLS{
+						{
+							Hosts: []string{"foo.example.org"},
+						},
+					},
+				},
+			}
+			ingressB := &networkingv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "ingress-2",
+					Namespace: shootNamespace.Name,
+					Labels: map[string]string{
+						v1beta1constants.LabelShootEndpointAdvertise: "true",
+					},
+				},
+				Spec: networkingv1.IngressSpec{
+					TLS: []networkingv1.IngressTLS{
+						{
+							Hosts: []string{"bar.example.org"},
+						},
+					},
+				},
+			}
+
+			Expect(botanist.SeedClientSet.Client().Create(ctx, ingressA)).To(Succeed())
+			Expect(botanist.SeedClientSet.Client().Create(ctx, ingressB)).To(Succeed())
+			items, err := botanist.GetIngressAdvertisedEndpoints(ctx)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(items).To(HaveLen(2))
+			Expect(items).To(Equal([]gardencorev1beta1.ShootAdvertisedAddress{
```

Comment:
What about using `HaveExactElements`? This way, you could remove the length check.